src/customFileParser/main.ts

```diff
@@ -1,81 +1,61 @@
+import { unpackInstancedMeshes, unpackMergedMeshes, unpackPrimitives } from './unpackGeometry/main';
 import Sector from './../Sector';
-import countRenderedGeometries from './countRenderedGeometries';
-import { SectorMetadata, GeometryIndexHandler, UncompressedValues, RenderedGeometryGroups }
-  from './sharedFileParserTypes';
 import CustomFileReader from './CustomFileReader';
-import { renderedGeometries, renderedGeometryToGroup } from './parserParameters';
+import SceneStats from './../SceneStats';
 
-function createSector(
-  sectorMetadata: SectorMetadata,
-  uncompressedValues: UncompressedValues,
-  geometryIndexHandlers: GeometryIndexHandler[]): Sector {
-
-  const counts = countRenderedGeometries(geometryIndexHandlers);
-  const renderedGeometryGroups: RenderedGeometryGroups = {};
-  renderedGeometries.forEach(renderedGeometry => {
-    const count = counts[renderedGeometry];
-    renderedGeometryGroups[renderedGeometry] = new renderedGeometryToGroup[renderedGeometry](count);
-  });
-
-  const groupNames = ['Box', 'Circle', 'ClosedCone', 'ClosedCylinder', 'ClosedEccentricCone', 'ClosedEllipsoidSegment',
-    'ClosedExtrudedRingSegment', 'ClosedGeneralCylinder', 'ClosedSphericalSegment', 'ClosedTorusSegment', 'Ellipsoid',
-    'ExtrudedRing', 'Nut', 'OpenCone', 'OpenCylinder', 'OpenEccentricCone', 'OpenEllipsoidSegment',
-    'OpenExtrudedRingSegment', 'OpenGeneralCylinder', 'OpenSphericalSegment', 'OpenTorusSegment', 'Ring', 'Sphere',
-    'Torus', 'TriangleMesh', 'InstancedMesh'];
-  groupNames.forEach(name => {
-    unpackGeometry(renderedGeometryGroups, geometryIndexHandlers, uncompressedValues, name);
-  });
-
-  const sectorObject = new Sector(sectorMetadata.sectorBBoxMin, sectorMetadata.sectorBBoxMax);
-  Object.keys(renderedGeometryGroups).forEach(renderedGeometryName => {
-    if (renderedGeometryName === 'TriangleMesh') {
-
-    } else if (renderedGeometryName === 'InstancedMesh') {
-
-    } else {
-      sectorObject.primitiveGroups.push(renderedGeometryGroups[renderedGeometryName]);
-    }
-  });
-
-  return sectorObject;
-}
-
-function parseManySectors(fileBuffer: ArrayBuffer): Sector {
+export default function parseCustomFile(fileBuffer: ArrayBuffer) {
   const fileReader = new CustomFileReader(fileBuffer);
-  const sectors: {[name: string]: any} = {};
-
-  let sectorStartLocation = 0;
-  let sectorByteLength = fileReader.readUint32();
-  let sectorMetadata = fileReader.readSectorMetadata(sectorByteLength);
-  const uncompressedValues = fileReader.readUncompressedValues();
-  let geometryIndexHandlers = fileReader.readSectorGeometryIndexHandlers(sectorStartLocation + sectorByteLength);
-  const rootSector = createSector(sectorMetadata, uncompressedValues, geometryIndexHandlers);
-  sectors[sectorMetadata.sectorId] = rootSector;
+  const idToSectorMap: {[name: string]: Sector} = {};
+  const sceneStats: SceneStats = {
+    numInstancedMeshes: 0,
+    numMergedMeshes: 0,
+  };
+  const mergedMeshMap: any = {};
```

Comment:
We have the type 
```
export interface InstancedMeshMap {
  [key: number]: InstancedMesh;
}
```
in `parsers/parseUtils.ts`. Not sure why we don't have one for merged meshes (my bad I suppose). Could you create a type there instead?

Reply:
We don't actually use the mergedMeshMap anywhere yet, so I'm not sure why you're initialising it in Parser.ts (as a todo?). Also it's currently 'const mergedMeshMap: InstancedMeshMap = {};' there. I'm taking out this line entirely.